cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaTopicConfigProvider.java

```diff
@@ -88,8 +146,10 @@ private void loadClusterConfigs(String clusterConfigsFile) throws FileNotFoundEx
 
   @Override
   public void configure(Map<String, ?> configs) {
-    _connectString = (String) configs.get(ExecutorConfig.ZOOKEEPER_CONNECT_CONFIG);
-    _zkSecurityEnabled = (Boolean) configs.get(ExecutorConfig.ZOOKEEPER_SECURITY_ENABLED_CONFIG);
+
+    KafkaCruiseControlConfig ccConfig = new KafkaCruiseControlConfig(configs);
+    _adminClient = KafkaCruiseControlUtils.createAdminClient(KafkaCruiseControlUtils.parseAdminClientConfigs(ccConfig));
+
```

Comment:
Rather than creating a new admin client here, can we use the existing admin client of `KafkaCruiseControl`? We can pass the existing admin client to the TopicConfigProvider upon configuration.
-- please see example usage in other pluggable component implementations.

Reply:
So that was my original intention but I thought that would break compatibility with the `TopicConfigProvider` and `CruiseControlConfigurable` interfaces. Are you suggesting we add a new `configure` method signiture to the `CruiseControlConfigurable` interface or somehow add the Admin client into the `Map<String,?> given to the existing `configure` method?

Also I was worried that as `TopicConfigProvider` is auto-closable and we are using a single admin client this may cause issues on shutdown (unless we make the `close()` method no-op in the new topic config provider)? But maybe that is not such a concern?

Reply:
Regarding the first point, we can achieve this w/o any changes in the API of `TopicConfigProvider` or `CruiseControlConfigurable` interfaces. I left a comment on how to do so with an existing example -- e.g. `AnomalyDetectorUtils#KAFKA_CRUISE_CONTROL_OBJECT_CONFIG`.

Regarding the second point, we should indeed make this a no-op close() as the shared admin client is already being closed last at `KafkaCruiseControl#shutdown()`.